examples/amm/src/state.rs

```diff
@@ -0,0 +1,8 @@
+use linera_sdk::views::{MapView, ViewStorageContext};
+use linera_views::views::{GraphQLView, RootView};
+
+#[derive(RootView, GraphQLView)]
+#[view(context = "ViewStorageContext")]
+pub struct Amm {
+    pub token_pool: MapView<String, u64>,
```

Comment:
I think that instead of having multiple tokens supported in an application instance, we should fix it to two. Every trade pair can then be a separate application instance running in parallel. The tokens can be specified when the application is created with global parameters (see the [Crowd Funding application](https://github.com/linera-io/linera-protocol/blob/main/examples/crowd-funding/src/lib.rs#L247) which has one token as input).

I'm wondering if we need any fields for the state at all :thinking: In theory we can query the respective Fungible Token application instances to get the AMM application's token balance every time we need it. That might be more correct, but likely less fuel efficient.

Note that multi-token pools are possible, but are more complicated (see [Balancer](https://medium.com/balancer-protocol/the-benefits-of-multi-token-pools-653eea3ef03a) for example).

Reply:
I like the idea of having no state at all (apart from, later, the unfilled standing bids).